include/boost/gil/extension/numeric/affine.hpp

```diff
@@ -108,6 +108,36 @@ boost::gil::matrix3x2<T> inverse(boost::gil::matrix3x2<T> m)
     return res;
 }
 
+//feature
+//rotate an image from its center point for -180 deg <= theta <= 180 deg
+//using consecutive affine transformations
+//As top-left corner of source image is taken as origin, idea is to rotate the source image about the origin and then translate the new image
+//according to the mentioned rotation angle, and to re-center and fit the image in the source dimensions 
+// preferably use type double for less precision errors
+template<typename T, typename F> boost::gil::matrix3x2<F> center_rotate(boost::gil::point<T> dims,F rads)
+{   
+    while(rads < F(-M_PI)) rads = rads + F(M_PI);
+    while(rads > F(M_PI)) rads = rads - F(M_PI);
+
+    F c_theta = std::abs(std::cos(rads));
+    F s_theta = std::abs(std::sin(rads));
+    
+    boost::gil::matrix3x2<F> translation(0,0,0,0,0,0);
+    if(rads >= 0) { translation.b = -s_theta; }
+    else { translation.c = -s_theta; }
+
+    if(std::abs(rads) > F(M_PI/2)) { translation.a = -c_theta, translation.d = -c_theta; }
+
+    return boost::gil::matrix3x2<F>::get_scale(s_theta * dims.y / dims.x + c_theta, s_theta * dims.x / dims.y + c_theta) *  //to fit inside given dimensions
+           boost::gil::matrix3x2<F>::get_translate(dims * translation) *  //re-center the image [top-left of image taken as (0,0)]
```

Comment:
Please, respect the line length limit
Perhaps add some intermediate results, then comments could go into separate lines, instead of squeezing everything into long expression of the `return` statement.

Reply:
I have re-written the code with exceeding length limits